### ISSUE TITLE
Fails to create a parameter file for arm template when encoded with UTF-8 BOM

### ISSUE THREAD

**Issue by kernst123** [microsoft/vscode-azurearmtools#721]
Does this occur consistently? Yes 
Repro steps:

1. from vs code create a new arm template (use snippet arm! )
2. from vs code select/create parameter file...
I receive the error below

Action: azurerm-vscode-tools.selectParameterFile
Error type: Error
Error Message: "e:\temp\armtemp\mytemplate.json" does not appear to be an Azure Resource Manager deployment template file.

Version: 0.10.0
OS: win32
OS Release: 10.0.17763
Product: Visual Studio Code
Product Version: 1.45.1
Language: en

<details>
<summary>Call Stack</summary>

```
Object.<anonymous> extension.bundle.js:1:596601
fulfilled extension.bundle.js:1:593686
```

</details>

**Comment by StephenWeatherford** [comment]
@kernst123 Thanks for entering!

Was the file saved before doing step 2?  If not, it's probably a duplicate of https://github.com/microsoft/vscode-azurearmtools/issues/666.

**Comment by kernst123** [comment]
Hi,

Yes it was saved before I invoked the command.
-- Kevin

**Comment by StephenWeatherford** [comment]
@neilpeterson This looks vaguely familiar but I can't find any dups.  Have you seen this behavior?  Thx.

**Comment by neilpeterson** [comment]
@StephenWeatherford nope this appears to be unique as compared to similar open items.

I was however able to reproduce the error / call stack by trying to create a parameters file from an invalid ARM template (removed the opening bracket).

@kernst123 can you share the ARM template that you using when attempting to create the parameter file?

Thanks

**Comment by kernst123** [comment]
Here is an example of the arm template it failed for.
I blanked out the tenant-id and the objectID so you will nee to put something relevant in for that.
My original version I did infact run and it created the key vault just to confirm that it is valid.
I hope this helps.
[test.json.zip](https://github.com/microsoft/vscode-azurearmtools/files/4664297/test.json.zip)

**Comment by kernst123** [comment]
Hmmm I thought I posted a zip file.  I don't see it above.  I will try again.
[test.json.zip](https://github.com/microsoft/vscode-azurearmtools/files/4664349/test.json.zip)

**Comment by neilpeterson** [comment]
Thanks @kernst123, I noticed that both tenant and object id did have values so removed the file from the issue. 

I am looking at it now, but unfortunately cannot reproduce the issue. I am able to create the parameter file without issue using the provided tempalte.

**Comment by kernst123** [comment]
Hmm it's  a tricky one.
Here is a screenshot showing the error.
![image](https://user-images.githubusercontent.com/40530719/82598866-51909780-9b79-11ea-83bd-150b95869f79.png)

Is there any logging I could enable on my end?
Or any version info that might be relevant that I could show you?

**Comment by StephenWeatherford** [comment]
I can repro with the attached, thanks (not sure why Neil can't).  We're having trouble with the UTF-8 BOM (the one on disk, not the text buffer).

**Comment by neilpeterson** [comment]
Ahhh, works fine on my Mac but does repo in Windows.

**Comment by StephenWeatherford** [comment]
Weird, I repro'd it with my Mac.

**Comment by kernst123** [comment]
As soon as I changed the encoding from utf8 bom to utf8 it started working.
thanks.
I changed vscode's default to utf8 from the old value utf8 bom.

**Comment by StephenWeatherford** [comment]
Fix released with 0.11.0
